dmapiclient/audit.py

```diff
@@ -39,6 +39,7 @@ class AuditTypes(Enum):
     send_clarification_question = "send_clarification_question"
     send_application_question = "send_application_question"
     answer_selection_questions = "answer_selection_questions"
+    agree_framework_variation = "agree_framework_variation"
```

Comment:
Also it's interesting that when a framework agreement is returned (which seems like just as, if not more, important than a contract variation) that we don't have a specific audit event for this. We are instead using the `supplier_update` event (as seen [here](https://github.com/alphagov/digitalmarketplace-api/blob/master/app/main/views/suppliers.py#L459). I think it seems sensible to have a more specific audit type like you are doing and maybe we should have done similar when a framework agreement is returned. @pcraig3